python/lsst/ts/externalscripts/auxtel/latiss_acquire_and_take_sequence.py

```diff
@@ -240,116 +256,191 @@ def set_metadata(self, metadata):
             gratings.add(grating)
         metadata.filter = f"{filters},{gratings}"
 
-    async def latiss_acquire(self, silent=True, doPointingModel=False):
+    async def latiss_acquire(self, doPointingModel=False):
 
         if doPointingModel:
-            targetPosition = boreSight
+            target_position = latiss_constants.boresight
         else:
-            targetPosition = sweetSpots[self.acq_grating]
+            target_position = latiss_constants.sweet_spots[self.acq_grating]
+
+        # get current filter/disperser
+        current_filter, current_grating, current_stage_pos = await self.latiss.get_setup()
+
+        # Check if a new configuration is required
+        if (self.acq_filter is not current_filter) and (self.acq_grating is not current_grating):
+            self.log.debug(f"Must load new filter {self.acq_filter } and grating {self.acq_grating}")
 
-        # Automatically accept calculated offset to sweetspot?
-        # if False this queries the user, which will fail badly on
-        # the script queue so make it always True
-        _alwaysAcceptMove = True
+            # Is the atspectrograph Correction in the ATAOS running?
+            corr = await self.atcs.rem.ataos.evt_correctionEnabled.aget(timeout=self.cmd_timeout)
+            if corr.atspectrograph:
+                # If so, then flush correction events for confirmation of
+                # corrections
+                self.atcs.rem.ataos.evt_atspectrographCorrectionStarted.flush()
+                self.atcs.rem.ataos.evt_atspectrographCorrectionCompleted.flush()
 
+        # Setup instrument and telescope
         await asyncio.gather(
-            self.attcs.slew_object(name=self.object_name, pa_ang=0, slew_timeout=240),
-            self.latiss.setup_atspec(grating=self.acq_grating, filter=self.acq_filter),
+            self.atcs.slew_object(name=self.object_name, rot=0, slew_timeout=240),
+            self.latiss.setup_instrument(grating=self.acq_grating, filter=self.acq_filter),
         )
 
-        success = False
-        iterNum = -1  # heinous to support max_acq_iter = 0 working TODO: fix this!
-        for iterNum in range(self.max_acq_iter):
-
-            retvals = await asyncio.gather(
-                findOffsetsAndMove(
-                    self.attcs,
-                    targetPosition,
-                    self.latiss,
-                    dataId=None,
-                    butler=self.butler,
-                    doMove=True,
-                    alwaysAcceptMove=_alwaysAcceptMove,
-                ),
+        # If ATAOS is running wait for adjustments to complete before
+        # moving on.
+        if corr.atspectrograph:
+            self.log.debug(f"Verifying LATISS configuration is incorporated into ATAOS offsets")
+            # If so, then flush correction events for confirmation of
+            # corrections
+            await self.atcs.rem.ataos.evt_atspectrographCorrectionStarted.aget(timeout=self.cmd_timeout)
```

Comment:
Are you sure you want to use `aget` here and in the following line? you are flushing the events above so I think you may want to use `next` with `flush=False`...

Reply:
uhh what's the difference?
Is it that if multiple events are received then `next` picks the one right after the flush whereas `aget` takes the most recent?

Reply:
Yes. So, if you get here before the new event is published you get the last one seen (with `aget`), even with the flush above. Whereas `next` will give you what ever was published after the flush.

Reply:
right, because `aget` and `get` aren't affected by flushes... good catch.